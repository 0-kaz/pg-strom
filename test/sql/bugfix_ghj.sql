---#
---#	GpuHashJoin Closed Issue Test Cases
---#

-- off to any join plan
set enable_hashjoin to off;
set enable_mergejoin to off;
set enable_nestloop to off;
-- off to any scan plan
set enable_indexscan to off;
set enable_bitmapscan to off;

set client_min_messages to warning;

-- #109
drop table ttest;
create table ttest(id integer,c1 real,c2 text);
alter table ttest alter c2 set storage external;
insert into ttest select generate_series(1,1),1000,repeat('a',3000)||generate_series(1,10);

select a.id,b.id,a.c1,b.c1,substring(a.c2,1,1),substring(b.c2,1,10) from ttest a inner join ttest b on a.id=b.id;

