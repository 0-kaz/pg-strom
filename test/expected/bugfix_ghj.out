---#
---#	GpuHashJoin Closed Issue Test Cases
---#
-- off to any join plan
set enable_hashjoin to off;
set enable_mergejoin to off;
set enable_nestloop to off;
-- off to any scan plan
set enable_indexscan to off;
set enable_bitmapscan to off;
set client_min_messages to warning;
-- #109
drop table ttest;
ERROR:  table "ttest" does not exist
create table ttest(id integer,c1 real,c2 text);
alter table ttest alter c2 set storage external;
insert into ttest select generate_series(1,1),1000,repeat('a',3000)||generate_series(1,10);
select a.id,b.id,a.c1,b.c1,substring(a.c2,1,1),substring(b.c2,1,10) from ttest a inner join ttest b on a.id=b.id;
 id | id |  c1  |  c1  | substring | substring  
----+----+------+------+-----------+------------
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
  1 |  1 | 1000 | 1000 | a         | aaaaaaaaaa
(100 rows)

